packages/repository/test/acceptance/has-many.relation.acceptance.ts

```diff
@@ -56,41 +57,48 @@ describe('HasMany relation', () => {
   });
 
   it('can patch many instances', async () => {
-    await createSamples();
-    await testPatch();
-
-    async function testPatch() {
-      const patchObject = {description: 'new order'};
-      const arePatched = await customerOrders.patch(patchObject);
-      // tslint:disable-next-line: no-unused-expression
-      expect(arePatched).to.equal(2);
-      const patchedItems = await customerOrders.find();
-      expect(patchedItems).to.have.length(2);
-      patchedItems.forEach(order => {
-        expect(order.description).to.eql('new order');
-        expect(order.customerId).to.eql(existingCustomerId);
-      });
-    }
+    await givenCustomerOrder({description: 'order 1', isDelivered: false});
+    await givenCustomerOrder({description: 'order 2', isDelivered: false});
+    const patchObject = {description: 'new order'};
+    const arePatched = await customerOrders.patch(patchObject);
+    // tslint:disable-next-line: no-unused-expression
+    expect(arePatched).to.equal(2);
+    const patchedData = _.map(await customerOrders.find(), d =>
+      _.pick(d, ['customerId', 'description', 'isDelivered']),
+    );
+    expect(patchedData).to.eql([
+      {
+        customerId: existingCustomerId,
+        description: 'new order',
+        isDelivered: false,
+      },
+      {
+        customerId: existingCustomerId,
+        description: 'new order',
+        isDelivered: false,
+      },
+    ]);
+  });
+
+  it('throws error when query tries to change the foreignKey', async () => {
+    await expect(
+      customerOrders.patch({customerId: existingCustomerId + 1}),
+    ).to.be.rejectedWith(/Field "customerId" cannot be changed!/);
   });
 
   it('can delete many instances', async () => {
-    await createSamples();
-    await testDelete();
-
-    async function testDelete() {
-      const deletedOrders = await customerOrders.delete();
-      // tslint:disable-next-line: no-unused-expression
-      expect(deletedOrders).to.equal(2);
-      const relatedOrders = await customerOrders.find();
-      // tslint:disable-next-line: no-unused-expression
-      expect(relatedOrders).to.be.empty;
-    }
+    await givenCustomerOrder({description: 'order 1'});
+    await givenCustomerOrder({description: 'order 2'});
+    const deletedOrders = await customerOrders.delete();
+    // tslint:disable-next-line: no-unused-expression
+    expect(deletedOrders).to.equal(2);
+    const relatedOrders = await customerOrders.find();
+    // tslint:disable-next-line: no-unused-expression
+    expect(relatedOrders).to.be.empty;
```

Comment:
Please note that `empty` is a function, see http://shouldjs.github.io/#assertion-empty

Please remove `no-unused-expression` override too.

Reply:
Good catch! Probably because I didn't call empty as a function, tslint was complaining unused expression. Removed those comments.